File: server/common/safe_transport.py
# Asegura que se recibe todo el payload - Short read handler
def safe_rcv(client_sock, total_length):
    """Recibe datos de forma segura, asegurando que se recibe todo el payload."""
    payload = b''
    while len(payload) < total_length:
        bytes_recv = client_sock.recv(total_length - len(payload))
        if not bytes_recv:
            raise ConnectionError("Conexión cerrada al leer payload")
        payload += bytes_recv
    return payload

# Asegura que se envía todo el paquete - Short write handler
def safe_send(client_sock, data):
    """Envía datos de forma segura, asegurando que se envía todo el paquete."""
    total_sent = 0
    while total_sent < len(data):
        bytes_sent = client_sock.send(data[total_sent:])
        if bytes_sent == 0:
            raise ConnectionError("Conexión cerrada al enviar payload")
        total_sent += bytes_sent

File: client/common/safe_transport.go
package common

import "net"

// Asegura que se envía todo el paquete - Short write handler
func SafeSend(conn net.Conn, data []byte) error {
	packet_send_len := 0
	for packet_send_len < len(data) {
		n, err := conn.Write(data[packet_send_len:])
		if err != nil {
			return err
		}
		packet_send_len += n
	}
	return nil
}

// Asegura que se recibe todo el paquete - Short read handler
func SafeRecv(conn net.Conn, length int) ([]byte, error) {
	buf := make([]byte, length)
	packet_recv_len := 0
	for packet_recv_len < length {
		n, err := conn.Read(buf[packet_recv_len:])
		if err != nil {
			return nil, err
		}
		packet_recv_len += n
	}
	return buf, nil
}
